core/state/database.go

```diff
@@ -182,40 +187,31 @@ func NewDatabase(db ethdb.Database) Database {
 // large memory cache.
 func NewDatabaseWithConfig(db ethdb.Database, config *trie.Config) Database {
 	return &cachingDB{
-		disk:                  db,
-		codeSizeCache:         lru.NewCache[common.Hash, int](codeSizeCacheSize),
-		codeCache:             lru.NewSizeConstrainedCache[common.Hash, []byte](codeCacheSize),
-		triedb:                trie.NewDatabaseWithConfig(db, config),
-		addrToPoint:           utils.NewPointCache(),
-		StorageProcessed:      map[common.Hash]bool{},
-		CurrentAccountAddress: map[common.Hash]*common.Address{},
-		CurrentSlotHash:       map[common.Hash]common.Hash{},
-		CurrentPreimageOffset: map[common.Hash]int64{},
+		disk:          db,
+		codeSizeCache: lru.NewCache[common.Hash, int](codeSizeCacheSize),
+		codeCache:     lru.NewSizeConstrainedCache[common.Hash, []byte](codeCacheSize),
+		triedb:        trie.NewDatabaseWithConfig(db, config),
+		addrToPoint:   utils.NewPointCache(),
 	}
 }
 
 // NewDatabaseWithNodeDB creates a state database with an already initialized node database.
 func NewDatabaseWithNodeDB(db ethdb.Database, triedb *trie.Database) Database {
 	return &cachingDB{
-		disk:                  db,
-		codeSizeCache:         lru.NewCache[common.Hash, int](codeSizeCacheSize),
-		codeCache:             lru.NewSizeConstrainedCache[common.Hash, []byte](codeCacheSize),
-		triedb:                triedb,
-		addrToPoint:           utils.NewPointCache(),
-		ended:                 triedb.IsVerkle(),
-		StorageProcessed:      map[common.Hash]bool{},
-		CurrentAccountAddress: map[common.Hash]*common.Address{},
-		CurrentSlotHash:       map[common.Hash]common.Hash{},
-		CurrentPreimageOffset: map[common.Hash]int64{},
+		disk:          db,
+		codeSizeCache: lru.NewCache[common.Hash, int](codeSizeCacheSize),
+		codeCache:     lru.NewSizeConstrainedCache[common.Hash, []byte](codeCacheSize),
+		triedb:        triedb,
+		addrToPoint:   utils.NewPointCache(),
 	}
 }
 
 func (db *cachingDB) InTransition() bool {
-	return db.started && !db.ended
+	return db.CurrentTransitionState != nil && db.CurrentTransitionState.started && !db.CurrentTransitionState.ended
```

Comment:
the assumption here is that if `CurrentTransitionState` is `nil`, then the transition hasn't started.